packages/common/src/scorekeeper/jobs/JobConfigs.ts

```diff
@@ -43,215 +44,113 @@ export enum JobNames {
-        await validatorPrefJobWithTiming(jobRunnerMetadata);
-      },
-      name: JobNames.ValidatorPref,
-      preventOverlap: true,
-    };
-
-    const locationStatsJobConfig: JobConfig = {
-      jobKey: "locationStats",
-      defaultFrequency: Constants.LOCATION_STATS_CRON,
-      jobFunction: async () => {
-        await locationStatsJobWithTiming(jobRunnerMetadata);
-      },
-      name: JobNames.LocationStats,
-      preventOverlap: true,
-    };
-
-    const nominatorJobConfig: JobConfig = {
-      jobKey: "nominator",
-      defaultFrequency: Constants.NOMINATOR_CRON,
-      jobFunction: async () => {
-        await nominatorJobWithTiming(jobRunnerMetadata);
-      },
-      name: JobNames.Nominator,
-      preventOverlap: true,
-    };
-
-    const blockDataJobConfig: JobConfig = {
-      jobKey: "block",
-      defaultFrequency: Constants.BLOCK_CRON,
-      jobFunction: async () => {
-        await blockJobWithTiming(jobRunnerMetadata);
-      },
-      name: JobNames.BlockData,
-      preventOverlap: true,
-    };
-
-    const mainScorekeeperJobConfig: JobConfig = {
-      jobKey: "scorekeeper",
-      defaultFrequency: Constants.SCOREKEEPER_CRON,
-      jobFunction: async () => {
-        await mainScorekeeperJob(jobRunnerMetadata);
-      },
-      name: JobNames.MainScorekeeper,
-      preventOverlap: true,
-    };
-
-    const executionJobConfig: JobConfig = {
-      jobKey: "execution",
-      defaultFrequency: Constants.EXECUTION_CRON,
-      jobFunction: async () => {
-        await executionJob(jobRunnerMetadata);
-      },
-      name: JobNames.Execution,
-      preventOverlap: true,
-    };
-
-    const cancelJobConfig: JobConfig = {
-      jobKey: "cancel",
-      defaultFrequency: Constants.CANCEL_CRON,
-      jobFunction: async () => {
-        await cancelJob(jobRunnerMetadata);
-      },
-      name: JobNames.Cancel,
-      preventOverlap: true,
-    };
-
-    const staleNominationJobConfig: JobConfig = {
-      jobKey: "stale",
-      defaultFrequency: Constants.STALE_CRON,
-      jobFunction: async () => {
-        await staleNominationJob(jobRunnerMetadata);
-      },
-      name: JobNames.StaleNomination,
-      preventOverlap: true,
-    };
-
-    return [
-      activeValdiatorJobConfig,
-      monitorJobConfig,
-      clearOfflineJobConfig,
-      validityJobConfig,
-      scoreJobConfig,
-      eraStatsJobConfig,
-      eraPointsJobConfig,
-      inclusionJobConfig,
-      sessionKeyJobConfig,
-      unclaimedEraJobConfig,
-      validatorPrefJobConfig,
-      locationStatsJobConfig,
-      nominatorJobConfig,
-      blockDataJobConfig,
-      mainScorekeeperJobConfig,
-      executionJobConfig,
-      cancelJobConfig,
-      staleNominationJobConfig,
-    ];
-  } catch (e) {
-    logger.error(`Error getting job configs:`, jobsLabel);
-    logger.error(JSON.stringify(e));
-    return [];
-  }
-};
+export const jobConfigs: JobConfig[] = [
+  {
+    jobKey: "activeValidator",
```

Comment:
does it make sense to have these jobKeys defined as constants or enums instead of hardcoded strings ?

Reply:
Yes, it absolutely makes sense. This question also related to the default cron strings, which are located in the constants.

Current state:
- Config holds information about both job keys and cron strings.
- Source code also holds information about the job keys and default cron strings. The source code assumes the config could be not valid.

The question is - why do we have defaults for the cron strings? If we trust the config, then we should have only one source of truth.

I suggest:
- Add an Enum containing all of the Job names (`jobKey`). It would be our source of truth.
- Validate the config structure (contains all jobs keys, cron string, isActive flag, etc.).
- Remove default cron strings from the source code.

This way we'd also simplify `JobConfig` type.

Instead of this👇 we'd need only mapping between `jobKey` and `jobFunction`:
```
  {
    jobKey: "activeValidator",
    defaultFrequency: Constants.ACTIVE_VALIDATOR_CRON,
    jobFunction: activeValidatorJobWithTiming,
    name: JobNames.ActiveValidator,
  },
 ```
 
 WDYT?